pkg/login/social/generic_oauth.go

```diff
@@ -107,12 +109,9 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 		s.log.Debug("Processing external user info", "source", data.source, "data", data)
 
 		if userInfo.Name == "" {
-			if data.Name != "" {
-				s.log.Debug("Setting user info name from name field")
-				userInfo.Name = data.Name
-			} else if data.DisplayName != "" {
-				s.log.Debug("Setting user info name from display name field")
-				userInfo.Name = data.DisplayName
+			userInfo.Name = s.extractUserName(data)
+			if userInfo.Name != "" {
+				s.log.Debug("Set user info name from extracted name", "name", userInfo.Name)
```

Comment:
I think that this debug message is redundant given that each special case in `extractUserName` is logged.